single_level.py

```diff
@@ -28,106 +28,47 @@ def get_note_types(resource):
 			note_types.append(note["jsonmodel_type"])
 	return note_types
 	
-def get_extent_number(resource):
-	extent_number = []
-	for extent in resource["extents"]:
-		if extent["number"]:
-			extent_number.append(extent["number"])
+def get_values(resource, array, value):
+	value_types = []
+	for item in resource[array]:
+		if item[value]:
+			value_types.append(item[value])
 		else:
-			extent_number.append("false")
-	return " , ".join(extent_number)
+			value_types.append("false")
+	return value_types
 	
-def get_language(resource):
+def get_single_value(resource, key):
 	d = resource
-	if 'language' in d:
-		return 'language'
+	if key in d:
+		return key
 	else:
 		return "false"
-
-def get_date_types(resource):
-	date_types = []
-	for date in resource["dates"]:
-		if date["date_type"]:
-			date_types.append("true")
-		else:
-			date_types.append("false")
-	return date_types
-
-def get_agent_roles(resource):
-	agent_roles = []
-	for agent in resource["linked_agents"]:
-		if agent["role"]:
-			agent_roles.append("true")
-		else:
-			agent_roles.append("false")
-	return " , ".join(agent_roles)
-	
+		
 def makeRow(resource):
 	global row
 	row = []
-	date_list = get_date_types(resource)
-	extent_list = get_extent_number(resource)
-	agent_list = get_agent_roles(resource)
+	extent = get_values(resource, "extents", "number")
+	date = get_values(resource, "dates", "date_type")
+	agent = get_values(resource, "linked_agents", "role")
+	language = get_single_value(resource, "language")
+	repository = get_single_value(resource, "repository")
+	required_values = extent, date, language, repository
 	notes_list = get_note_types(resource)
 	row.append(resource["title"].encode("utf-8"))
 	row.append(resource["id_0"])
 	row.append(resource["publish"])
```

Comment:
I think you might be able to replace these lines with calls to the `get_single_value()` function.